src/main/java/com/streamr/client/utils/EncryptionUtil.java

```diff
@@ -0,0 +1,47 @@
+package com.streamr.client.utils;
+
+import com.streamr.client.exceptions.InvalidGroupKeyException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.xml.bind.DatatypeConverter;
+import java.security.SecureRandom;
+
+public class EncryptionUtil {
+    private static final Logger log = LogManager.getLogger();
+    private static final SecureRandom SRAND = new SecureRandom();
+
+    public static String encrypt(byte[] plaintext, SecretKey groupKey) {
+        try {
+            Cipher encryptCipher = Cipher.getInstance("AES/CTR/NoPadding");
```

Comment:
I was wondering about performance and reuse of the `Cipher` instance. Quick googling gave http://tutorials.jenkov.com/java-cryptography/cipher.html:

> Initializing a Java Cipher instance is an expensive operation. Therefore it is a good idea to reuse Cipher instances. Luckily, the Cipher class was designed with reuse in mind.

> When you call the doFinal() method on a Cipher instance, the Cipher instance is returned to the state it had just after initialization. The Cipher instance can then be used to encrypt or decrypt more data again.